### ISSUE TITLE
Resizing bracket elements

### ISSUE THREAD

**Issue by g0nP** [teijo/jquery-bracket#73]
Is it possible to increase the width of each Team box?

**Comment by kpagcha** [comment]
I am also interested in this. The label is way too small for regular team names.

**Comment by lordone24** [comment]
I agree. Let me know the method please.

**Comment by teijo** [comment]
Hi,

At the moment you'd need to modify the library a bit to achieve that.

Here is the spot where the full width of the bracket is calculated: https://github.com/teijo/jquery-bracket/blob/master/src/jquery.bracket.ts#L1150

 `140` is sum of the width of the box (100px) and the margin (40px) for the connector. I.e. if you want to double the box width, the value should be `240`.

Then you need to modify these two styles (`.team` and `.round`) to your desired width, e.g. `200px` in the above example:
https://github.com/teijo/jquery-bracket/blob/master/src/jquery.bracket.sass#L74
https://github.com/teijo/jquery-bracket/blob/master/src/jquery.bracket.sass#L59

You could also just override that CSS from outside without modifying the library, but the JavaScript modification cannot be skipped for now.

**Comment by kpagcha** [comment]
I am applying this after bracket generation and that helps customizing a bit the width:

`// width multiplier
var w_mult = 1.5;

var round_w = $('.round').width();
$('.round').width(round_w * w_mult);

var team_w = $('.team').width();
$('.team').width(team_w* w_mult);

var n_rounds = $('.round').length;
var connector_w = $('.connector').width();
var round_w = $('.round').width();
$('.jQBracket').width(n_rounds * round_w + n_rounds * connector_w * 2);

// width multiplier for the score slot
var w_mult_score = 1.35;
var score_w = $('.score').width();
$('.score').width(score_w * w_mult_score);`

I am going to try doing the same for the height, but it will probably prove more diffucult.

The width modification doesn't seem to be very hard to implement, does it?

**Comment by teijo** [comment]
Modifying the size from outside is a bit dangerous if there are any sync issues, i.e. you need to first draw the bracket to see what size it is so you can do the calculation based on it.

Implementing the resizing support into the library isn't that hard. Maybe I'll look at it next. The reason I've been kinda putting it off is that there are a lot of parameters that people might want to customise, not just team width, and in general that should be done with CSS. However CSS (especially for older browsers) isn't really suitable for this kind of generated layouts.

**Comment by kpagcha** [comment]
> Modifying the size from outside is a bit dangerous if there are any sync issues, i.e. you need to first draw the bracket to see what size it is so you can do the calculation based on it.

I am not sure I follow. Why would it be dangerous? What kind of sync issues? The example you said, I am not really sure what you mean with that. I **am** modifying the structure after the brackets are generated.

**Comment by Jabe64** [comment]
Hi!

If you are looking at this next, please implement Responsive Design as available on http://www.gotbracket.com

As you said on https://github.com/teijo/jquery-bracket/issues/26#issuecomment-191494965
> "GotBracket uses actually a simple JavaScript resize listener and calculates suitable zoom CSS value based on window size."

**Comment by teijo** [comment]
@kpagcha 
> Why would it be dangerous? What kind of sync issues?

It's more of a hunch and I'm not 100% sure about this, but I think it may be possible to create the DOM elements in JS and immediately read their size before a layout / rendering has been done for them. In that case you might get invalid or zero width / height as the browser hasn't yet had the chance to properly determine them.

In any case, if you render something and immediately resize it, you'll get flickering. Only good way to do it, is to do it correctly on the first pass, when you have all the numbers in your hands inside the library. In ideal world CSS would do all this.

> Could you share where in the plugin is the connector position calculated so I can figure out the right modifier to reposition them accordingly?

Go through all `height()` and `width()` functions as well as `height` and `width` styles :)

**Comment by teijo** [comment]
Some initial size support in master 8e2986a325a92498b3b8b871b063b4ced5f10c88:
```
teamWidth: number;
scoreWidth: number;
roundMargin: number;
matchMargin: number;
```

**Comment by kpagcha** [comment]
Thank you, great work! Really looking forward to the following additions. By the way, could you update the distro folder with the new versions? I am trying to compile the resources but I'm having issues, particularly with the sass file.

**Comment by teijo** [comment]
> I am trying to compile the resources but I'm having issues, particularly with the sass file.

What's the problem? It "should" work by just running `npm install` and then `grunt` (or `npm start` for auto-compile when changes happen).

> By the way, could you update the distro folder with the new versions?

I don't want to clutter the version history with intermediate compiled versions.

>  the label still has a fixed width

Yea the input box and the container are still the old size. Will fix those too.

**Comment by teijo** [comment]
I recommend just getting `node` so you can use the proper toolchain: https://nodejs.org/en/

It's hard to get by without those if you want to do anything more complicated than plain JS.

**Comment by teijo** [comment]
Published 0.9.0. You can try out the interactive demo at http://www.aropupu.fi/bracket/#resize

**Comment by teijo** [comment]
@Jabe64 
> If you are looking at this next, please implement Responsive Design as available on http://www.gotbracket.com

I'm not yet sure if and how it should be part of the library as scaling might be really specific for each UI design. I.e. maybe you'd have a pan instead of zoom. Could maybe add better support for those things if necessary and someone comes up with what's needed.

If you want, you can make a new feature ticket for this.
